utils/src/debug_panic.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021-2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Panic in debug builds or log the message in release builds.
+/// Note that unlike the standard `panic` macro this one requires at least one argument.
+#[macro_export]
+macro_rules! debug_panic_or_log {
+    ($($arg:tt)+) => {
+        if cfg!(debug_assertions) {
+            panic!($($arg)*);
+        } else {
+            logging::log::error!("CRITICAL - debug build would panic: {}", format!($($arg)*));
```

Comment:
This is really a stupid thing to comment... but please either make it `CRITICAL:` so that it matches others. I usually grep for `CRITICAL:` or `CRITICAL ERROR:`. I'd rather change everything to the former.